File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        max_len = 0
        cur_len = 0

        for num in nums_set:
            if num - 1 not in nums_set:
                cur_num = num
                cur_len = 1
                while cur_num + 1 in nums_set:
                    cur_num += 1
                    cur_len += 1

                max_len = max(max_len, cur_len)

        return max_len






        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
import java.util.HashSet;

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) return 0;

        HashSet<Integer> set = new HashSet<>();
        for (int n : nums) set.add(n);

        int maxLen = 0;

        for (int n : set) {
            // start a sequence only if n-1 doesn't exist (n is a sequence head)
            if (!set.contains(n - 1)) {
                int curr = n;
                int len = 1;

                while (set.contains(curr + 1)) {
                    curr++;
                    len++;
                }
                if (len > maxLen) maxLen = len;
            }
        }

        return maxLen;
    }
}
